src/Extension/MemberExtension.php

```diff
@@ -31,95 +27,29 @@ class MemberExtension extends Extension implements PermissionProvider
      */
     public function updateCMSFields(FieldList $fields)
     {
-        $fields->removeByName(['LoginSessions', 'SessionManagerField']);
+        $fields->removeByName('LoginSessions');
+
+        $member = $this->getOwner();
+        if (!$member->exists()) {
+            // We're creating a new member
+            return;
+        }
 
-        if (!$this->owner->exists() || !$this->currentUserCanViewSessions()) {
-            return $fields;
+        $session = $member->LoginSessions()->First();
+        if (!$session || !$session->canView()) {
+            // We're assuming that the first session permission are representative of the other sessions
```

Comment:
This is a bit of an assumption, but it seems like a pretty safe one to make. Can't imagine what sort of crazy uses case would require you to be able to see some `LoginSession` for a Member, but not all of them.